lib/screens/signup/signup_repository.dart

```diff
@@ -22,3 +28,42 @@ class FakeSignupRepository implements SignupRepository {
     }
   }
 }
+
+class APISignupRepository implements SignupRepository {
+  final String classTag = "APILoginRepository";
+  @override
+  Future<void> signup(
+      String firstName, String lastName, String email, String mobileNumber, String password) async {
+    final String tag = classTag + "login";
+    http.Response response;
+    try {
+      response = await sl.get<http.Client>().post(
+        S.registerUrl,
+        body: <String, dynamic>{
+          S.firstnameKey: firstName.toString(),
+          S.lastnameKey: lastName.toString(),
+          S.emailKey: email.toString(),
+          S.phoneKey: mobileNumber.toString(),
+          S.passwordKey: password.toString()
+        },
+      );
+    } catch (e) {
+      Log.e(tag: tag, message: "NetworkError:" + e.toString());
+      throw NetworkException();
+    }
+
+    if (response.statusCode == 201) {
+      try {
+        Log.d(tag: tag, message: "Signup Successful ");
```

Comment:
Not really for **debugging**, but for **information**
```suggestion
        Log.i(tag: tag, message: "Signup Successful ");
```